mmv1/third_party/terraform/services/eventarc/resource_eventarc_google_api_source_test.go

```diff
@@ -0,0 +1,466 @@
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_eventarc_google_api_source.primary", plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+			{
+				ResourceName:            "google_eventarc_google_api_source.primary",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"annotations", "google_api_source_id", "labels", "location", "terraform_labels"},
+			},
+			{
+				Config: testAccEventarcGoogleApiSource_unset(context),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_eventarc_google_api_source.primary", plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+			{
+				ResourceName:            "google_eventarc_google_api_source.primary",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"annotations", "google_api_source_id", "labels", "location", "terraform_labels"},
+			},
+		},
+	})
+}
+
+// Sets up an initial project containing a GoogleApiSource with CMEK connected
+// to a MessageBus in the same project
+func testAccEventarcGoogleApiSource_full(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_project" "project" {
+  project_id      = "tf-test%{random_suffix}"
+  name            = "tf-test%{random_suffix}"
+  org_id          = "%{org_id}"
+  billing_account = "%{billing_account}"
+  deletion_policy = "DELETE"
+}
+
+resource "time_sleep" "wait_create_project" {
+  create_duration = "60s"
+  depends_on      = [google_project.project]
+}
+
+resource "google_project_service" "compute" {
+  project    = google_project.project.project_id
+  service    = "compute.googleapis.com"
+  depends_on = [time_sleep.wait_create_project]
+}
+
+resource "google_project_service" "servicenetworking" {
+  project   = google_project.project.project_id
+  service   = "servicenetworking.googleapis.com"
+  depends_on = [google_project_service.compute]
+}
+
+resource "google_project_service" "kms" {
+  project    = google_project.project.project_id
+  service    = "cloudkms.googleapis.com"
+  depends_on = [google_project_service.servicenetworking]
+}
+
+resource "google_project_service" "eventarc" {
+  project    = google_project.project.project_id
+  service    = "eventarc.googleapis.com"
+  depends_on = [google_project_service.kms]
+}
+
+resource "time_sleep" "wait_enable_service" {
+  create_duration = "60s"
+  depends_on      = [google_project_service.eventarc]
+}
+
+resource "google_kms_key_ring" "keyring" {
+  name       = "keyring"
+  location   = "%{region}"
+  project    = google_project.project.project_id
+  depends_on = [google_project_service.kms]
+}
+
+resource "google_kms_crypto_key" "key" {
+  name     = "key1"
+  key_ring = google_kms_key_ring.keyring.id
+}
+
+resource "google_project_service_identity" "eventarc_sa" {
```

Comment:
This test immediately failed in GA, as `google_project_service_identity` is beta-only. Is there a way to test this without it, or should the update test be beta-only? @tommyreddad @melinath

Reply:
Ugh, yeah, that's unfortunate. This is a requirement for testing updates to message_bus, so either we don't test updating that field or we don't test updates in GA.